### ISSUE TITLE
NullPointerException while compute accuracy with ComputeModelStatistics

### ISSUE THREAD

**Issue by ttpro1995** [microsoft/SynapseML#736]
Version 
```
com.microsoft.ml.spark:mmlspark_2.11:jar:0.18.1
spark= 2.4.3
scala=2.11.12
```

data (csv with header) https://gist.github.com/ttpro1995/69051647a256af912803c9a16040f43a 

download data and save as csv file, put into folder `/data/public/HIGGS/higgs.test.predictioncsv` 
```
val data = spark.read.option("header","true").option("inferSchema", "true").csv("/data/public/HIGGS/higgs.test.predictioncsv")
```

Schema
```
root
 |-- label: double (nullable = true)
 |-- prediction: double (nullable = true)
```

Code

```
import com.microsoft.ml.spark.train.ComputeModelStatistics
val metricsCompute = new ComputeModelStatistics().setLabelCol("label").setScoresCol("prediction").setEvaluationMetric("accuracy")

val result_metrics = metricsCompute.transform(data)
```

Exception 

```
java.lang.NullPointerException
  at org.apache.spark.sql.Column.<init>(Column.scala:135)
  at org.apache.spark.sql.Column$.apply(Column.scala:38)
  at org.apache.spark.sql.functions$.col(functions.scala:90)
  at com.microsoft.ml.spark.train.ComputeModelStatistics.selectAndCastToDF(ComputeModelStatistics.scala:256)
  at com.microsoft.ml.spark.train.ComputeModelStatistics.selectAndCastToRDD(ComputeModelStatistics.scala:265)
  at com.microsoft.ml.spark.train.ComputeModelStatistics.predictionAndLabels$lzycompute$1(ComputeModelStatistics.scala:99)
  at com.microsoft.ml.spark.train.ComputeModelStatistics.predictionAndLabels$1(ComputeModelStatistics.scala:95)
  at com.microsoft.ml.spark.train.ComputeModelStatistics.transform(ComputeModelStatistics.scala:124)
  ... 47 elided
```

**Comment by imatiach-msft** [comment]
@ttpro1995 thank you for the detailed repro.  Will take a look at this when I get a chance.  Can you please post how you read the "data" variable just so I'm not missing any of the parts the repros the issue?  My first guess is there might be some missing values in the dataset.

**Comment by ttpro1995** [comment]
code that work in python sklearn (so, data is not broken)

```
from sklearn.metrics import  accuracy_score
import pandas as pd
data = pd.read_csv("/data/public/HIGGS/higgs.test.predictioncsv/part-00000-2cf6c2c8-173c-49c1-ae78-9dfe697f3a0f-c000.csv")
data.head()
accuracy_score(data["label"], data["prediction"])
```

**Comment by ttpro1995** [comment]
Another case with error using different data

Notebook on zepl.com 
https://www.zepl.com/viewer/notebooks/bm90ZTovL2hhaGF0dHByby9iMzBiYjJiNzZkMGI0YWVlYTQwMjMwY2JiMzNkYWE0MC9ub3RlLmpzb24 

Export to zeppelin: 
https://gist.github.com/ttpro1995/8e6227039b2593041267180a1d81efa7 

Data using in notebook:  

binary.train  https://github.com/microsoft/LightGBM/blob/master/examples/binary_classification/binary.train

binary.test https://github.com/microsoft/LightGBM/blob/master/examples/binary_classification/binary.test

**Comment by imatiach-msft** [comment]
@ttpro1995 oh sorry, I just saw the problem now, you are setting the scores column but not the scored labels column, which is required if you are trying to compute all metrics:
https://github.com/Azure/mmlspark/blob/master/src/main/scala/com/microsoft/ml/spark/train/ComputeModelStatistics.scala#L99
I think the error message needs to be improved

**Comment by ttpro1995** [comment]
I don't know what each function, each set col do.
i.e: what setLabelCol, setScoredLabelsCol, setScoresCol ... these name are similar and confusing.

I suggest there should me more detail in [documentation](https://mmlspark.blob.core.windows.net/docs/1.0.0-rc1/scala/index.html#com.microsoft.ml.spark.train.ComputeModelStatistics) . Including: meaning of each parameter, default value (I have to new a object in code, then use get function to know the default value),

**Comment by imatiach-msft** [comment]
@ttpro1995 I think the column name convention is similar, but maybe not exactly, in most ML platforms, at least in scikit-learn, apache spark and ML.NET.  The labelCol is the true labels or original label from the dataset, the ScoredLabelsCol is the label assigned by the classifier, and the ScoresCol is the raw prediction, for example the distance from the separating hyperplane for the support vector machine classifier.  There's also usually a probability col, which is not used in ComputeModelStatistics, although you could pass it as the input to the ScoresCol too (it's used for the AUC metric).

**Comment by imatiach-msft** [comment]
@ttpro1995 "I suggest there should be more detail in documentation . Including: meaning of each parameter, default value (I have to new a object in code, then use get function to know the default value),"
totally agree, the documentation could definitely be improved here.  thanks for pointing that out.

**Comment by ranadeepsingh** [comment]
Resolved by #2633, merged into `master` as `7fd1a0b86762ca54c0a4ef7306e1d0f8827fb613`.

`ComputeModelStatistics` now validates configured label, scored-label, and score columns before metadata or Catalyst access. Missing, ambiguous, empty, unsupported, and invalid vector inputs produce actionable parameter-specific errors instead of the reported `NullPointerException`. The fix also aligns runtime/schema behavior and preserves configured parameters through copy, persistence, and generated Python wrappers.

Thanks for the report and documentation feedback.
